sources/api/migration/migrator/src/main.rs

```diff
@@ -328,8 +422,131 @@ where
 ///
 /// The given data store is used as a starting point; each migration is given the output of the
 /// previous migration, and the final output becomes the new data store.
-fn run_migrations<P1, P2>(
+fn run_migrations<P1, P2, S>(
+    repository: &tough::Repository<'_, tough::FilesystemTransport>,
     direction: Direction,
+    migrations: &[S],
+    source_datastore: P1,
+    new_version: &Version,
+    migrations_rundir: P2,
+) -> Result<PathBuf>
+where
+    P1: AsRef<Path>,
+    P2: AsRef<Path>,
+    S: AsRef<str>,
+{
+    // We start with the given source_datastore, updating this after each migration to point to the
+    // output of the previous one.
+    let mut source_datastore = source_datastore.as_ref();
+    // We create a new data store (below) to serve as the target of each migration.  (Start at
+    // source just to have the right type; we know we have migrations at this point.)
+    let mut target_datastore = source_datastore.to_owned();
+    // Any data stores we create that aren't the final one, i.e. intermediate data stores, will be
+    // removed at the end.  (If we fail and return early, they're left for debugging purposes.)
+    let mut intermediate_datastores = HashSet::new();
+
+    for migration in migrations {
+        let migration = migration.as_ref();
+        // get the migration from the repo
+        let lz4_bytes = repository
+            .read_target(migration)
+            .context(error::LoadMigration { migration })?
+            .context(error::MigrationNotFound { migration })?;
+
+        // deflate with an lz4 decoder read
+        let mut reader = lz4::Decoder::new(lz4_bytes).context(error::Lz4Decode { migration })?;
+
+        // TODO - remove this use of the filesystem when we add pentacle
+        let exec_path = migrations_rundir.as_ref().join(&migration);
+        {
+            let mut f = OpenOptions::new()
+                .write(true)
+                .create(true)
+                .truncate(true)
+                .open(&exec_path)
```

Comment:
This is exactly [`File::create`](https://doc.rust-lang.org/stable/std/fs/struct.File.html#method.create).

Reply:
It went away with pentacle anyway.